### DISCUSSION TITLE
Webpage shows null when trying to access it.

### DISCUSSION THREAD

**Discussion by Electric-Dragon** [deta/cloud-docs#227] [Help] [answered]
Hello,
I am using a `NodeJS` Micro and it is configured this way:

```
require('dotenv').config();
const express = require("express");
const path = require('path');

const app = express();
const port = 7001;

const http = require('http');
var server = http.createServer(app);
const io = require("socket.io")(server);

let socket;

io.on('connection', function (socket_) {
    socket = socket_;
    console.log("Connected succesfully to the socket ...");
});

app.use(express.static(path.join(__dirname + '/public')));
app.use(express.urlencoded({extended:true}));

app.get("/", function(req, res) {
    res.sendFile(__dirname + "/html/home.html");
});

server.listen(process.env.PORT || port, function() {
    console.log(`Server started on http://localhost:${port}`);
})

module.export = server;
```

Instead of showing the home page, this is what is shown:
<img width="1440" alt="Bildschirmfoto 2022-05-30 um 9 58 13 AM" src="https://user-images.githubusercontent.com/45729391/170917230-14aa0e99-eb9a-4385-80e3-379c4e463e44.png">

**Comment by abdelhai** [comment, marked as answer]
1. Deta Micros don't support long-living sockets. [see #12](https://docs.deta.sh/docs/micros/about#important-notes)
2. the server component we will run **must** be called `app` so your export expression should be: `module.export = app`.

**Reply by Electric-Dragon** [reply]
Okay...will it work if I do `export default`?

Also, is there any other way to use socket.io?

**Reply by abdelhai** [reply]
No, we require the exported object to be called `app`.  
No, as per my link. Reading the "Important Note" will give you an understanding of how Micros work.

**Reply by Electric-Dragon** [reply]
@abdelhai Will Deta work if I add the `.deta` folder in an existing nodejs application?

**Reply by aavshr** [reply]
hi @Electric-Dragon it might work but we do not recommend doing so, it might lead to unexpected behavior. The `.deta` folder should only be manipulated by the cli. May I ask what you need to do this for?

**Reply by Electric-Dragon** [reply]
@aavshr I created a micro...but then the cli created a new directory, so I moved the `.deta` into the root folder of the project and deleted the created directory

**Reply by aavshr** [reply]
@Electric-Dragon ok, but please be aware that this might lead to unexpected behavior.

**Reply by Electric-Dragon** [reply]
@aavshr Can you suggest a way to use deta in an existing nodeJS application then?

**Reply by aavshr** [reply]
@Electric-Dragon you can use `deta new` command. More here: https://docs.deta.sh/docs/cli/commands#deta-new

**Reply by Electric-Dragon** [reply]
@aavshr I tried it but the website still shows `null`

**Reply by Electric-Dragon** [reply]
I fixed it...apart from the `module.exports=app` there was an issue in the environment variables
